### ISSUE TITLE
Client-only component in production. TypeError: Cannot read properties of null (reading 'parentNode')

### ISSUE THREAD

**Issue by kingyue737** [nuxt/nuxt#28070]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.20.3
- Nuxt Version:     3.12.3
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      devtools, modules, ssr
- Runtime Modules:  @vueuse/nuxt@10.11.0
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-bf56vz-76m3ye

### Describe the bug

Enter the reproduction and waiting for `npm run generate && npm run preview`.

> Sometimes `npm run generate` fails on stackblitz with error ` ERROR  Command failed with exit code 13: npx serve ./public `. Try `npm run generate` and `npm run preview` again or build the reproduction project locally. I don't know why, maybe a bug of stackblitz.

See the error in console:
```
TypeError: Cannot read properties of null (reading 'parentNode')
```

### Additional context

No issue in dev mode.

I meet several of the same errors in my projects in production, all related to client-only components. Currently, I only succeeded in making a minimal reproduction of this one. They may have the same underlying problem.

Maybe related to #27999

**Comment by jd1378** [comment]
I am getting the same error for my Client only component only when I try to change routes, it renders fine on initial load
I use `.client` with a `.server` pair

**Comment by agracia-foticos** [comment]
TypeError: Cannot read properties of null (reading 'parentNode')
    at parentNode (BV4ETHL4.js:17:654)
    at ns.$ [as fn] (BV4ETHL4.js:13:32824)
    at ns.run (BV4ETHL4.js:9:1517)
    at V.y.update (BV4ETHL4.js:13:33530)
    at ft (BV4ETHL4.js:13:46)
    at Zr (BV4ETHL4.js:13:1802)
    at Zr (BV4ETHL4.js:13:1891)

**Comment by agracia-foticos** [comment]
Related ??? https://github.com/nuxt/nuxt/issues/13309

**Comment by systempower999** [comment]
vue 3.4.35 solves the problem

**Comment by kingyue737** [comment]
> vue 3.4.35 solves the problem

No, I tried vue 3.4.35. Still the same error

**Comment by agracia-foticos** [comment]
https://stackblitz.com/edit/nuxt-starter-fynqph in PROD mode must be fail if navigates between 1 and 2 link

**Comment by time4dev** [comment]
> > vue 3.4.35 solves the problem
> 
> No, I tried vue 3.4.35. Still the same error

"nuxt": "^3.12.4", "@vue/server-renderer": "^3.4.35", "vue": "^3.4.35" This solved the problem for me.

**Comment by kingyue737** [comment]
Fixed in Nuxt v3.13.0
